benefits/oauth/views.py

```diff
@@ -85,71 +85,6 @@ def login(request):
     return result
 
 
-@decorator_from_middleware(FlowUsesClaimsVerificationSessionRequired)
-def authorize(request):
-    """View implementing OIDC token authorization."""
-    flow = session.flow(request)
-
-    oauth_client_result = _oauth_client_or_error_redirect(request, flow)
-
-    if hasattr(oauth_client_result, "authorize_access_token"):
-        # this looks like an oauth_client since it has the method we need
-        oauth_client = oauth_client_result
-    else:
-        # this does not look like an oauth_client, it's an error redirect
-        return oauth_client_result
-
-    logger.debug("Attempting to authorize OAuth access token")
-    token = None
-    exception = None
-
-    try:
-        token = oauth_client.authorize_access_token(request)
-    except Exception as ex:
-        exception = ex
-
-    if token is None:
-        logger.warning("Could not authorize OAuth access token")
-        exception = Exception("oauth_client.authorize_access_token returned None")
-
-    if exception:
-        analytics.error(request, message=str(exception), operation="authorize_access_token")
-        sentry_sdk.capture_exception(exception)
-        return redirect(routes.OAUTH_SYSTEM_ERROR)
-
-    logger.debug("OAuth access token authorized")
-
-    # oauth_token_authorized will be stored in the user's session to mark the user as authorized
-    oauth_token_authorized = True
-
-    # We store the returned claim in case it can be used later in eligibility verification.
-    flow_claims = flow.claims_request.claims_list
-    stored_claims = []
-
-    error_claim = {}
-
-    if flow_claims:
-        userinfo = token.get("userinfo")
-
-        if userinfo:
-            for claim in flow_claims:
-                claim_value = userinfo.get(claim)
-                # the claim comes back in userinfo like { "claim": "1" | "0" }
-                claim_value = int(claim_value) if claim_value else None
-                if claim_value is None:
-                    logger.warning(f"userinfo did not contain: {claim}")
-                elif claim_value == 1:
-                    # if userinfo contains our claim and the flag is 1 (true), store the *claim*
-                    stored_claims.append(claim)
-                elif claim_value >= 10:
-                    error_claim[claim] = claim_value
-
-    session.update(request, oauth_authorized=oauth_token_authorized, oauth_claims=stored_claims)
```

Comment:
Since we lost this `session.update()` call with `oauth_authorized=True`, Benefits doesn't "know" the user is signed in with the IdG / IdP.

I think we'd need to update the hook(s) to set this value. Or like I've been saying, just generally fix our session :sweat_smile:

Reply:
Oh man, I was 100% under the impression that I had added that to the hooks in #2767 because that's what I was going to do when writing https://github.com/cal-itp/benefits/pull/2767#issuecomment-2737873588. But now I see with the back and forth about dependent tickets, I never did make that change.

Thanks for catching that, and I can add it here.

I absolutely want to get the session refactored to follow your new pattern soon, but I also feel like it's too much to try and do while also introducing the change of switching over to `cdt_identity`. It would feel less overwhelming to me to get things working first with `cdt_identity` and then finish it out with refactoring existing behavior.